### PR TITLE
Limit expression nesting level to avoid stack overflows

### PR DESCRIPTION
**Pull request by djc**
This is triggered by the fuzzer, see also https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=62759.

### CODE REVIEW

**Comment by Kijewski** [comment]
Wouldn't same be needed for Nodes, too?

```jinja2
{%if true%}{%if true%}{%if true%}{%if true%}{%if true%}{%if true%}{%if true%}{%if true%}...
```

**Comment by djc** [comment]
Probably? Was going to wait until the fuzzer complains.

File: askama_parser/src/expr.rs [hunk 1]
```diff
@@ -15,11 +15,12 @@ use super::{
...
     ( $name:ident, $inner:ident, $op:expr ) => {
-        fn $name(i: &'a str) -> IResult<&'a str, Self> {
-            let (i, left) = Self::$inner(i)?;
+        fn $name(i: &'a str, level: Level) -> IResult<&'a str, Self> {
+            let level = level.nest(i)?;
+            let (i, left) = Self::$inner(i, level)?;
             let (i, right) = many0(pair(
                 ws(tag($op)),
-                Self::$inner,
+                |nested| Self::$inner(nested, level),
```
Comment (Kijewski) [inline askama_parser/src/expr.rs:23]: I was confused for a moment what `nested` was. I think the identifier `i` would be better. We use it throughout all the code and it is used in all of nom's examples for the input string.

Comment (djc) [reply askama_parser/src/expr.rs]: Fair enough, renamed them.

**Review by Kijewski — APPROVED** [review]

**Pull request merged**